pyscript.core/docs/README.md

```diff
@@ -0,0 +1,341 @@
+    print(document.currentScript.id == my_target.id)
+</script>
+```
+
+Not only this is helpful to crawl the surrounding *DOM* or *HTML*, every script will also have a `target` property that will point either to the element reachable through the `target` attribute, or it lazily creates once a companion element that will be appended right after the currently executing *script*.
+
+Please read the [Terminology](#terminology) **target** dedicated details to know more.
+
+  </div>
+</details>
+
+<details>
+  <summary><strong>XWorker</strong></summary>
+  <div>
+
+With or without access to the `document`, every (*non experimental*) interpreter will have defined, at the global level, a reference to the `XWorker` "_class_" (it's just a *function*!), which goal is to enable off-loading heavy operations on a worker, without blocking the main / UI thread (the current page) and allowing such worker to even reach the `document` or anything else available on the very same main / UI thread.
+
+```html
+<script type="micropython">
+    # XWorker is globally defined
+    print(XWorker != None)
+</script>
+```
+
+Please read the [XWorker](#xworker) dedicated section to know more.
+
+  </div>
+</details>
+
+
+## How Events Work
+
+Inspired by the current [HTML Standard](https://html.spec.whatwg.org/multipage/webappapis.html#event-handlers):
+
+> the event handler is exposed through a name, which is a string that always starts with "_on_" and is followed by the name of the event for which the handler is intended.
+
+We took a similar approach, replacing that `on` prefix with whatever *interpreter* or *custom type* is available on the page, plus a *dash* `-` to avoid clashing with standards:
+
+```html
+<script type="micropython">
+    def print_type(event, double):
+        # logs "click 4"
+        print(f"{event.type} {double(2)}")
+</script>
+<button micropython-click="print_type(event, lambda x: x * 2)">
+    print type
+</button>
+```
+
+If this example felt a bit verbose, be ensured custom types would work the same:
+
+```html
+<!-- ℹ️ - requires py-script custom type -->
+<button py-click="print(event.type)">
+    print type
+</button>
+```
+
+What is important to understand about *events* in PyScript is that the text within the attribute is executed just like any other inline or external content is, through the very same *interpreter*, with the notably extra feature that the `event` reference is made temporarily available as *global* by *core*.
+
+This really reflects how otherwise native Web inline events handlers work and we think it's a great feature to support ... *but*:
+
+ * if your script runs *asynchronously* the `event` might be gone on the main / UI thread and by that time any of its `event.stopPropagation()` or `event.preventDefault()` goodness will be problematic, as too late to be executed
+ * if your *interpreter* is *experimental*, or incapable of running *synchronous* events, the `event` reference might be less useful
```

Comment:
I think unless we're planning to release/encourage users to release experimental interpreters, we can probably cut this.

Reply:
I agree that, written as is, it can add more confusion. I.e., what should the user do? Are we saying "don't use this feature for experimental interpreters"?